### PR TITLE
Use dao/bulk loader for gene panel matrix import

### PR DESCRIPTION
**Pull request by zheins**
Signed-off-by: zheins <zackheins@gmail.com>

# What? Why?
Imports of gene panel matrix files is horribly slow due to not using the Dao classes with cached records and the mysql bulk loader. For our internal clinical datasets, just importing this one datatype can take upwards of 20 minutes, and it's only going to get worse as the data sets grow.

By making these changes, the time of import of this datatype for one of our largest studies was decreased from ~20m to ~1m.

Changes proposed in this pull request:
- Use the Dao classes for looking up records
- Use the mysql bulk loader for loading sample profile records.

# Checks
- [x] Runs on Heroku.
- [x] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Follows the [Google Style Guide](https://github.com/google/styleguide).
- [x] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)
- [x] If this is a feature, the PR is to rc. If this is a bug fix, the PR is to
  hotfix.

# Notify reviewers
@n1zea144 @sheridancbio @angelicaochoa

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGenePanelProfileMap.java [hunk 1]
```diff
@@ -125,27 +126,31 @@ public void importData() throws Exception {
...
+                if (DaoSampleProfile.sampleExistsInGeneticProfile(sample.getInternalId(), profileIds.get(i))) {
+                    samplesToDelete.add(sample.getInternalId());
+                    profilesToDelete.add(profileIds.get(i));                           
+                }
             }
         }
+        ProgressMonitor.setCurrentMessage("Deleting necessary records from sample_profile.");
+        DaoSampleProfile.deleteRecords(samplesToDelete, profilesToDelete);
+        ProgressMonitor.setCurrentMessage("Loading gene panel profile matrix data to database..");
+        MySQLbulkLoader.flushAll();
```
Comment (sheridancbio) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGenePanelProfileMap.java]: I don't understand how the MySQLbulkLoader is connected to intercept calls to DaoSampleProfile.addSampleProfile() ... somehow it is injected into the normal connection to data sources? I don't see where.

File: core/src/main/java/org/mskcc/cbio/portal/dao/DaoSampleProfile.java [hunk 2]
```diff
@@ -276,4 +276,22 @@ public static void deleteAllRecords() throws DaoException {
...
     }
+    
+    public static void deleteRecords(List<Integer> sampleIds, List<Integer> profileIds) throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoSampleProfile.class);
+            for (int i =0; i < sampleIds.size(); i++) {
+                String sql = "DELETE FROM sample_profile WHERE sample_id = '" + String.valueOf(sampleIds.get(i)) + "' and genetic_profile_id = '" + String.valueOf(profileIds.get(0)) + "';";
```
Comment (zheins) [inline core/src/main/java/org/mskcc/cbio/portal/dao/DaoSampleProfile.java]: Oops, thanks yeah you're right.

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGenePanelProfileMap.java (hunk 1 above)
Comment (zheins) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGenePanelProfileMap.java]: Woops, I lost the code that did this when I realized there was an existing Dao class for SampleProfile.

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGenePanelProfileMap.java [hunk 3]
```diff
@@ -125,27 +126,31 @@ public void importData() throws Exception {
...
-                    map.put("panelId", genePanel.getInternalId());
-                    genePanelRepository.insertGenePanelSampleProfileMap(map);
+                GenePanel genePanel = DaoGenePanel.getGenePanelByStableId(data.get(i));
+                if (genePanel != null) {
+                    DaoSampleProfile.addSampleProfile(sample.getInternalId(), profileIds.get(i), genePanel.getInternalId());
                 }
                 else {
                     ProgressMonitor.logWarning("No gene panel exists: " + data.get(i));
                 }
+                if (DaoSampleProfile.sampleExistsInGeneticProfile(sample.getInternalId(), profileIds.get(i))) {
```
Comment (zheins) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGenePanelProfileMap.java]: The mybatis implementation was doing an update if I remember correctly.

**Pull request merged**